src/tools/openapi.ts

```diff
@@ -145,16 +166,18 @@ Only use the OpenAPI tool if you need to communicate to external servers.`;
     url.pathname = join(url.pathname, path);
     // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
     const headers: { [key: string]: string } = { Accept: "application/json" };
-    if (this.apiKey) {
-      headers["Authorization"] = `Bearer ${this.apiKey}`;
+    if (this.options.apiKey) {
+      headers["Authorization"] = `Bearer ${this.options.apiKey}`;
     }
```

Comment:
create body in advance, pass it to the emitter

Reply:
I'm not sure what to do here.  Would you explain a little more?  Thanks!

Reply:
```ts
 const options = {
    ...this.option.fetchOptions,
    body: !isEmpty(input.body) ? input.body : undefined,
    method: input.method.toLowerCase(),
    headers: { Accept: "application/json", ...this.option.fetchOptions?.headers, },
    
  }
 await this.emitter.emit("beforeFetch", { url, optionsl });
    try {
      const response = await fetch(
        url.toString(),
        options,
      );
      const text = await response.text();
      const output = new OpenAPIToolOutput(response.status, response.statusText, text);
      await this.emitter.emit("afterFetch", { data: output, url });
      return output;
    }
```

and change type of `fetchOptions` from `any` to `RequestInit`

Reply:
Thanks!!